### PR TITLE
[23] Add data collection pipeline for NIH clinical trials

### LINKED ISSUE #23: Pipeline to collect NIH Clinical Trials

### PR DESCRIPTION
**Pull request by Jack-0-0**
Closes #23 

This PR adds:
* `nih_raw_clinical_trials` partitioned CSV dataset to the data catalog
* Pipeline and nodes for collecting all NIH clinical trials data from `clinicaltrials.gov` (around 480k trials)

### CODE REVIEW

File: conf/base/catalog.yml
Comment (ampudia19) [inline conf/base/catalog.yml:1]: I'm starting to really enjoy this idea of importing "baseline" data formats, and then adapting them. It's v clean!

File: conf/base/parameters.yml [hunk 1]
```diff
@@ -79,4 +79,13 @@ gtr:
       param_requests: ${.._param_requests}
       label: projects
 
+# National Institutes of Health (nih) configuration 
+nih:
+  # Data collection parameters
+  data_collection:
+    clinical_trials:
+      download_url: https://classic.clinicaltrials.gov/AllAPIJSON.zip
+      zip_save_path: data/00_temp/nih_ct.zip
```
Comment (ampudia19) [inline conf/base/parameters.yml]: I'm a bit torn about using `00_tmp` - everything downloaded / requested should be in `01_raw`, going by what I can see in Kedro Slack. Also, I think it should be `00_tmp`, at least that's how the folder is labelled in S3.

File: src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/nodes.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    logger.info("Downloading ZIP file from %s", zip_url)
+    response = requests.get(zip_url, timeout=30)
+    response.raise_for_status()
+    zip_save_path = Path(zip_save_path)
+    zip_save_path.parent.mkdir(parents=True, exist_ok=True)
+    with open(zip_save_path, "wb") as file:
+        file.write(response.content)
+
+    logger.info("Downloaded ZIP file to %s", zip_save_path)
+    return zip_save_path
```
Comment (ampudia19) [inline src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/nodes.py]: You may want to change the name of the Path() object for it to be distinct from the path str argument. It makes it cleaner.

**Comment by Jack-0-0** [comment]
@ampudia19 I will update this to use namespaces as you suggested in #28

File: src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/pipeline.py
Comment (ampudia19) [inline src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/pipeline.py:1]: Here you can instead do:
```
    return pipeline(
        [
            node(
                func=collect_and_save_nih_clinical_trials,
                inputs={
                    "nih_ct_download_url": "params:download_url",
                    "nih_ct_zip_save_path": "params:zip_save_path",
                    "nih_ct_extract_path": "params:extract_path",
                },
                outputs="raw",
            ),
        ],
        namespace="nih.data_collection.clinical_trials"
    )
```
for the same functionality, but less bloated strings.

**Review by ampudia19 — APPROVED** [review]
Good stuff, minor comments!

File: src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/pipeline.py
Comment (Jack-0-0) [reply src/alphafold_impact/pipelines/data_collection_nih_clinical_trials/pipeline.py:1]: This is much cleaner. Thank you!

**Pull request merged**